js/ui/codingManager.js

```diff
@@ -80,11 +81,20 @@ const CodingManager = new Lang.Class({
     },
 
     addBuilderWindow: function(actor) {
+        let tracker = Shell.WindowTracker.get_default();
+        let correspondingApp = tracker.get_window_app(actor.meta_window);
+        let isSpeedwagonForBuilder = (
+            Shell.WindowTracker.is_speedwagon_window(actor.meta_window) &&
+            correspondingApp &&
+            correspondingApp.get_id() === 'org.gnome.Builder.desktop'
+        );
+
         if (!global.settings.get_boolean('enable-behind-the-screen'))
```

Comment:
That check should be done first in that function. Otherwise we will do the checks above for every app no matter the CodeView feature is enabled or not.